test/Upgrades.test.ts

```diff
@@ -0,0 +1,178 @@
+        this.Treasury = await ethers.getContractFactory('TreasuryInsurance')
+        this.Rewards = await ethers.getContractFactory('Rewards')
+        this.MockERC20 = await ethers.getContractFactory('MockERC20')
+        this.MockUnsafe = await ethers.getContractFactory('MockUnsafe')
+
+        this.timelock = <TimelockController>(
+            await this.TimelockController.deploy()
+        )
+        await this.timelock.deployed()
+        await this.timelock.initialize(
+            1,
+            [this.admin.address],
+            [this.admin.address]
+        )
+
+        this.votesOracle = <VotesOracle>(
+            await upgrades.deployProxy(this.VotesOracle, {kind: 'transparent'})
+        )
+        await this.votesOracle.deployed()
+
+        this.bit = <MockERC20>await this.MockERC20.deploy()
+        await this.bit.deployed()
+        await this.bit['initialize(string,string,uint256)'](
+            'BIT',
+            'BIT',
+            SUPPLY
+        )
+
+        this.governance = <Governance>await upgrades.deployProxy(
+            this.Governance,
+            {
+                kind: 'transparent',
+                initializer: false
+            }
+        )
+        await this.governance.deployed()
+
+        this.treasury = <TreasuryInsurance>(
+            await upgrades.deployProxy(
+                this.Treasury,
+                [this.governance.address, this.timelock.address],
+                {kind: 'transparent'}
+            )
+        )
+        await this.treasury.deployed()
+
+        await this.governance.initialize(
+            this.bit.address,
+            this.timelock.address,
+            this.votesOracle.address,
+            this.treasury.address
+        )
+        await this.timelock.grantRole(PROPOSER_ROLE, this.governance.address)
+        await this.timelock.grantRole(EXECUTOR_ROLE, this.governance.address)
+
+        this.rewards = <Rewards>(
+            await upgrades.deployProxy(
+                this.Rewards,
+                [
+                    this.governance.address,
+                    this.timelock.address,
+                    this.treasury.address,
+                    this.bit.address,
+                    REWARD_PER_VOTE
+                ],
+                {kind: 'transparent'}
+            )
+        )
+        await this.rewards.deployed()
+    })
+
+    it('Governance: Successful safe upgrade', async function () {
+        await expect(
+            upgrades.upgradeProxy(this.governance.address, this.Governance)
+        ).to.be.fulfilled
+        expect(await this.governance.votingDelay()).to.equal(1)
+        expect(await this.governance.votingPeriod()).to.equal(5)
+        expect(await this.governance.rolesList(0)).to.equal(TREASURY_ROLE)
+        expect(await this.governance.rolesList(1)).to.equal(DEVELOPER_ROLE)
+        expect(await this.governance.rolesList(2)).to.equal(LEGAL_ROLE)
+    })
+
+    it('Governance: Non-successful safe upgrade', async function () {
```

Comment:
suggestion: add comparison of implementation addresses.

This is to ensure the implementation contract has actually changed, provides a more complete test.